src/metabase/models/database.clj

```diff
@@ -101,10 +101,16 @@
     (let [all-users-group  (perms-group/all-users)
           non-magic-groups (perms-group/non-magic-groups)
           non-admin-groups (conj non-magic-groups all-users-group)]
-      ;; All other permissions are set at the table-level, so they are added when individual tables are synced
-      (data-perms/set-database-permission! all-users-group database :perms/native-query-editing :yes)
-      (doseq [group non-magic-groups]
-        (data-perms/set-database-permission! group database :perms/native-query-editing :no))
+      ;; We only set native-query-editing and manage-database permissions here, because they are only ever set at the
+      ;; database-level. Perms which can have table-level granularity are set in the `define-after-insert` hook for
+      ;; tables.
+      (if (:is_audit database)
+        (doseq [group non-admin-groups]
+          (data-perms/set-database-permission! group database :perms/native-query-editing :no))
+        (do
+          (data-perms/set-database-permission! all-users-group database :perms/native-query-editing :yes)
+          (doseq [group non-magic-groups]
+            (data-perms/set-database-permission! group database :perms/native-query-editing :no))))
```

Comment:
(this applies to the table change too): IIUC, When we add more permissions, we will need to make sure they're least-permissive in here. I wonder if we can build this behavior off of data-perms/Permissions so it happens automatically. That way we wouldn't have to touch these when adding perms.

Reply:
I agree in principle, but in https://www.notion.so/metabase/Change-the-default-access-of-new-added-databases-889d36bf79ce4a798a93148a8b992c0f we're going to be changing default DB perms to be a bit more complicated. So I think we might want to keep it explicit for now, and after we implement that project we can see how much stuff we can automatically infer from the permissions config